src/Storages/MergeTree/MergeTreeReadPoolBase.cpp

```diff
@@ -141,20 +141,7 @@ void MergeTreeReadPoolBase::fillPerPartInfos(const Settings & settings)
         MergeTreeReadTaskInfo read_task_info;
 
         read_task_info.data_part = part_with_ranges.data_part;
-
-        const auto & data_part = read_task_info.data_part;
-        /// TODO(ab): Hold parent part in RangesInDataParts during projection analysis
-        if (data_part->isProjectionPart())
-        {
-            read_task_info.parent_part = data_part->storage.getPartIfExists(
-                data_part->getParentPartName(),
-                {MergeTreeDataPartState::PreActive, MergeTreeDataPartState::Active, MergeTreeDataPartState::Outdated});
-
-            if (!read_task_info.parent_part)
-                throw Exception(ErrorCodes::LOGICAL_ERROR, "Did not find parent part {} for projection part {}",
-                            data_part->getParentPartName(), data_part->getDataPartStorage().getFullPath());
```

Comment:
Let's still have this check when `data_part->isProjectionPart() == true`?

Reply:
Oh sure.